### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by Uniqcoda**
Screenshot 2019-05-09 at 11 44 23 AM

### CODE REVIEW

File: method_files/alternatingCase.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+String.prototype.alternatingCase = function () {
+  let regex = /([a-z])/;
+  let array = this.split('');
+  let newArray = [];
+  // start by inversing the case of the first character
+  newArray.push(array[0].inverseCase());
+  for (let index = 1; index < array.length; index++) {
+    // for other characters, inverse their cases based on the case of the most recent character
+    if (regex.test(newArray[index-1])) {
```
Comment (mogbeyi-david) [inline method_files/alternatingCase.js]: Please remove the else statement from this block of logic

File: method_files/convertCharacter.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+function convertCharacter(character) {
+  let charCode = character.charCodeAt();
+  // check if character is lower case and subtract 32 from its code to convert to upper case
+  if (charCode > 96 && charCode < 123) {
+    return character = String.fromCharCode(charCode - 32);
+  } else if (charCode > 64 && charCode < 91) {
+    // check if character is upper case and add 32 to its code to convert to lower case
+    return character = String.fromCharCode(charCode + 32);
+  } else {
```
Comment (mogbeyi-david) [inline method_files/convertCharacter.js]: I do not think you need this else statement in this logic. Please review and refactor.

File: method_files/fromCurrency.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// A String method to convert a string of currency to numbers
+String.prototype.fromCurrency = function () {
+  let regex = /,/g;
+  while (regex.test(this)) {
+    // search for and replace all the commas with empty string
+    newNum = this.replace(regex, '');
```
Comment (mogbeyi-david) [inline method_files/fromCurrency.js]: This variable is implicitly declared, please declare properly.

File: method_files/hasVowels.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// A String method to check if a string has vowels
+String.prototype.hasVowels = function () {
+  let regex = /[aeiou]/g;
+  if (regex.test(this)) return true;
```
Comment (mogbeyi-david) [inline method_files/hasVowels.js]: NICE!

File: method_files/inverseCase.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// A String method that inverses upper case to lower case and vice-versa
+let toUpper = require('../method_files/toUpper');
+let toLower = require('../method_files/toLower');
+
+String.prototype.inverseCase = function () {
+  let regex1 = /([a-z])/;
```
Comment (mogbeyi-david) [inline method_files/inverseCase.js]: Please add a comment to explain what pattern this regex is matching

File: method_files/inverseCase.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  let regex2 = /([A-Z])/;
```
Comment (mogbeyi-david) [inline method_files/inverseCase.js]: Please add a comment to explain what pattern this regex is matching. Also, please confirm that you cannot collapse the two regular expressions into one by ignoring case-sensitivity.

File: method_files/isDigit.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+// A String method to check if a string is only one digit
+String.prototype.isDigit = function () {
+  let regex = /^[0-9]$/;
+  if (regex.test(this)) return true;
```
Comment (mogbeyi-david) [inline method_files/isDigit.js]: THUMBS UP!!!

File: method_files/isQuestion.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+// A String method to check if a string is a question
+String.prototype.isQuestion = function () {
+  let regex = /.+\?$/;
```
Comment (mogbeyi-david) [inline method_files/isQuestion.js]: Please add a comment to explain what the regex is matching.

File: method_files/toCurrency.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+// A String method to convert a string of numbers to currency, by adding commas
+String.prototype.toCurrency = function () {
+  // split the string in two, separated by the decimal point
+  let splitNumber = this.split('.');
+  let wholeNumber = splitNumber[0];
+  let decimalNumber = splitNumber.length > 1 ? '.' + splitNumber[1] : '';
+  let regex = /(\d+)(\d{3})/g;
```
Comment (mogbeyi-david) [inline method_files/toCurrency.js]: Please add a comment here to explain what this regex is matching.

File: method_files/toCurrency.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+  while (regex.test(wholeNumber)) {
+    // replace the whole number with a string that has a comma separating every# digits
+    wholeNumber = wholeNumber.replace(regex, '$1' + ',' + '$2');
```
Comment (mogbeyi-david) [inline method_files/toCurrency.js]: Please explain to me what $1 and $2 is?

File: method_files/alternatingCase.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
+// A String method that returns a string after alternating the case of its characters
+// import the toUpper, toLower and inverseCase methods in order to use them
+let toUpper = require('../method_files/toUpper');
+let toLower = require('../method_files/toLower');
+let inverseCase = require('../method_files/inverseCase');
+
+String.prototype.alternatingCase = function () {
+  let regex = /([a-z])/;
```
[the remark this answers is not in the document]
Comment (Uniqcoda) [reply method_files/alternatingCase.js]: done!

File: method_files/alternatingCase.js (hunk 1 above)
Comment (Uniqcoda) [reply method_files/alternatingCase.js]: done!

File: method_files/convertCharacter.js (hunk 2 above)
Comment (Uniqcoda) [reply method_files/convertCharacter.js]: I have refactored it

File: method_files/inverseCase.js (hunk 6 above)
Comment (Uniqcoda) [reply method_files/inverseCase.js]: The case is used as a conditional there cannot be ignored.

**Pull request closed without merging**